File: api/proxy/src/main/java/su/plo/voice/api/proxy/player/VoiceProxyPlayer.java
package su.plo.voice.api.proxy.player;

import org.jetbrains.annotations.NotNull;
import su.plo.slib.api.proxy.player.McProxyPlayer;
import su.plo.voice.api.server.player.VoicePlayer;

/**
 * Represents a voice API for the proxy player.
 */
public interface VoiceProxyPlayer extends VoicePlayer {

    @NotNull McProxyPlayer getInstance();
}

File: proxy/velocity/src/main/kotlin/su/plo/voice/velocity/extension/VoiceProxyPlayer.kt
package su.plo.voice.velocity.extension

import com.velocitypowered.api.proxy.Player
import su.plo.voice.api.proxy.player.VoiceProxyPlayer

/**
 * Gets a Velocity [Player] of [VoiceProxyPlayer].
 *
 * @return The Velocity player.
 */
fun VoiceProxyPlayer.asVelocityPlayer(): Player =
    instance.getInstance()
